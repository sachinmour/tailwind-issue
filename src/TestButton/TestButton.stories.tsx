import React from "react";
import TestButton from "./TestButton";

export default {
  title: "TestButton",
};

export const WithBar = () => <TestButton foo="bar" />;

export const WithBaz = () => <TestButton foo="baz" />;
