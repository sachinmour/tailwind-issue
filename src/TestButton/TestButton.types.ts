export interface TestButtonProps {
  foo: string;
}
