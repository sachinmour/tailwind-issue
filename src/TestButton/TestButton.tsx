import React from "react";
import clsx from "clsx";
import { TestButtonProps } from "./TestButton.types";

const TestButton: React.FC<TestButtonProps> = ({ foo }) => (
  <div
    data-testid="TestButton"
    className={clsx(
      "block bg-white text-left space-y-1 py-2 w-full",
      "sm:w-1/2 sm:space-y-0",
      "md:w-1/3 md:space-x-6",
      "dark:bg-gray-700"
    )}
  >
    {foo}
  </div>
);

export default TestButton;
