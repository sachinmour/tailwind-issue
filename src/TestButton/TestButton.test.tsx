/**
 * @jest-environment jsdom
 */
import React from "react";
import { render } from "@testing-library/react";
import TestButton from "./TestButton";
import { TestButtonProps } from "./TestButton.types";
import "jest-styled-components";

describe("Test Component", () => {
  let props: TestButtonProps;

  beforeEach(() => {
    props = {
      foo: "bar",
    };
  });

  const renderComponent = () => render(<TestButton {...props} />);

  it("should render foo text correctly", () => {
    props.foo = "harvey was here";
    const { getByTestId } = renderComponent();

    const component = getByTestId("TestButton");

    expect(component).toMatchSnapshot();
  });
});
