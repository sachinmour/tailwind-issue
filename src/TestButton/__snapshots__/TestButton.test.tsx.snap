// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component should render foo text correctly 1`] = `
<div
  class="block bg-white text-left space-y-1 py-2 w-full sm:w-1/2 sm:space-y-0 md:w-1/3 md:space-x-6 dark:bg-gray-700"
  data-testid="TestButton"
>
  harvey was here
</div>
`;
